source/advanced/testing-tango-using-docker.rst

```diff
@@ -0,0 +1,139 @@
+level: advanced; target: Tango developer
+
+
+Using Tango docker containers
+=============================
+
+In this section we describe how one can test newly developed tango device server using docker [link] containers.
+
+Tango docker containers provide a lightweight solution for deploying tango.
+
+To get info on how to install docker on your machine please refer to the docker documentation [link].
+
+Once docker is installed one can pull docker images with pre-installed tango.
+
+On the command line:
+
+:command:`docker pull tangocs/mysql`
+[further commands]
+
+
+Prerequisites
+-------------
+
+ - Docker engine [link]
+
+Tango docker containers
+-----------------------
+
+Tango provides two docker containers: mysql container with Tango scheme and tango-cs container with minimal set of Tango servers.
+
+These containers can be found in the corresponding repositories:
+https://github.com/tango-controls/docker-mysql
+
+[inject Dockerfile]
+
+https://github.com/tango-controls/tango-cs
+
+[inject Dockerfile]
+
+[header] Tango docker stack
+
+Typical tango stack looks like this:
+
+|image/tango-stack.png|
+
+The best way to setup the whole stack is to use docker compose [link]
+
+Here is an example of docker-compose.yml:
+
+.. code-block:: yaml
+
+    version: '2'
+    services:
+      tango-db:
+        image: tangocs/mysql:9.2.2
+        ports:
+         - "9999:3306"
+        environment:
+         - MYSQL_ROOT_PASSWORD=root
+      tango-cs:
+        image: tangocs/tango-cs:9
+        ports:
+         - "10000:10000"
+        environment:
+         - TANGO_HOST=localhost:10000
+         - MYSQL_HOST=tango-db:3306
+         - MYSQL_USER=tango
+         - MYSQL_PASSWORD=tango
+         - MYSQL_DATABASE=tango
+        depends_on:
+    - tango-db
+
+:command:`docker-compose up`
+
+[output]
+
+Once docker containers are up and running one can access mysql on localhost:9999 and the tango host on localhost:10000
+
+For instance, one can start jive (assuming it is installed on the system) [link to jive] :command:`TANGO_HOST=localhost:10000 jive`
+
+[img?]
+
+[header] Tango docker stack for Tango REST API
+
+One can setup Tango docker stack for Tango REST API [reference] as well. Currently only mtangorest.server [link] provides docker container that can be used together with Tango docker containers:
```

Comment:
Put in the Tango REST API document an anchor:

.. _tango_rest_api:

Your text..

Then use :ref:`tango_rest_api` to provide a link